app/controllers/admin/project_steps_controller.rb

```diff
@@ -14,4 +14,27 @@ def show
     @step = ProjectStep.find(params[:id])
     redirect_to admin_loan_path(@step.project_id, anchor: 'timeline')
   end
+
+  def update
+    @step = ProjectStep.find(params[:id])
+    if @step.update_with_translations(project_step_params, translations_params(@step.permitted_locales))
+      render :partial => "/admin/loans/timeline/project_step", :locals => {step: @step, mode: :show}
+    else
+      render :partial => "/admin/loans/timeline/project_step", :locals => {step: @step, mode: :edit}
+    end
+  end
+
+  private
+  def project_step_params
+    params.require(:project_step).permit(:is_finalized, :scheduled_date, :completed_date, :step_type_value)
+  end
+
+  def translations_params(locales)
+    translation_params_list = locales.reduce([]) { |res, l|
+      res = res + ["locale_#{l}".to_sym, "details_#{l}".to_sym, "summary_#{l}".to_sym]; res
+    }
```

Comment:
I believe the above could be rewritten as:
```
tp = locales.map { |l| %I(locale_#{l} details_#{l} summary_#{l}) }
```

Furthermore, it seems like perhaps similar code may be repeated all over the place in other controllers using translations. In which case, should consider moving this into a generic function in a concern that accepts a list of locales and a list of attrib prefixes.